pkg/joinserver/joinserver.go

```diff
@@ -389,28 +389,43 @@ func (js *JoinServer) HandleJoin(ctx context.Context, req *ttnpb.JoinRequest, au
 				cc = nil
 			}
 
-			var networkCryptoService cryptoservices.Network
-			if req.SelectedMacVersion.UseNwkKey() && dev.RootKeys != nil && dev.RootKeys.NwkKey != nil {
-				// LoRaWAN 1.1 and higher use a NwkKey.
+			// The root keys are used according to the following table:
+			//
+			//  Has NwkKey | Activation | Network uses | Application uses
+			//  ---------- | ---------- | ------------ | ----------------
+			//  No         | 1.0.x      | AppKey       | AppKey
+			//  Yes        | 1.0.x      | NwkKey       | NwkKey
```

Comment:
This is the new case; NwkKey should be used here.

The rest is mostly renaming by using NwkKey instead of AppKey for network layer crypto. That follows LoRaWAN 1.1 terminology. When there's no NwkKey and the device activates as 1.0.x, AppKey is filled in for NwkKey.